libs/animation/animation.ts

```diff
@@ -1,173 +1,624 @@
+        constructor() {
+            this.nodes = [];
+            this.lastNode = -1;
         }
 
-        _init() {
-            if (!animations) {
-                animations = [];
-                game.eventContext().registerFrameHandler(scene.ANIMATION_UPDATE_PRIORITY, () => {
-                    animations.forEach(anim => anim.update());
-                });
+        private static generateNode(p0: Point, command: string, args: number[], metadata: { pathStart: Point, lastNode: PathNode }): PathNode {
+            // Sets the start point for the path (that can be changed using moveTo)
+            let node: PathNode;
+            switch (command) {
+                case "M": { // M x y
+                    const p1 = new Point(args[0], args[1]);
+                    node = new MoveTo(p1);
+                    break;
+                }
+                case "m": { // m dx dy
+                    const p1 = new Point(p0.x + args[0], p0.y + args[1]);
+                    node = new MoveTo(p1);
+                    break;
+                }
+                case "L": { // L x y
+                    const p1 = new Point(args[0], args[1]);
+                    node = new LineTo(p0, p1);
+                    break;
+                }
+                case "l": { // l dx dy
+                    const p1 = new Point(p0.x + args[0], p0.y + args[1]);
+                    node = new LineTo(p0, p1);
+                    break;
+                }
+                case "H": { // H x
+                    const p1 = new Point(args[0], p0.y);
+                    node = new LineTo(p0, p1);
+                    break;
+                }
+                case "h": { // h dx
+                    const p1 = new Point(p0.x + args[0], p0.y);
+                    node = new LineTo(p0, p1);
+                    break;
+                }
+                case "V": { // V y
+                    const p1 = new Point(p0.x, args[0]);
+                    node = new LineTo(p0, p1);
+                    break;
+                }
+                case "v": { // v dy
+                    const p1 = new Point(p0.x, p0.y + args[0]);
+                    node = new LineTo(p0, p1);
+                    break;
+                }
+                case "Q": { // Q x1 y1 x2 y2
+                    const p1 = new Point(args[0], args[1]);
+                    const p2 = new Point(args[2], args[3]);
+                    node = new QuadraticCurveTo(p0, p1, p2);
+                    break;
+                }
+                case "q": { // q dx1 dy1 dx2 dy2
+                    const p1 = new Point(p0.x + args[0], p0.y + args[1]);
+                    const p2 = new Point(p0.x + args[2], p0.y + args[3]);
+                    node = new QuadraticCurveTo(p0, p1, p2);
+                    break;
+                }
+                case "T": { // T x2 y2
+                    let lastControlPoint: Point;
+                    if (metadata.lastNode instanceof QuadraticCurveTo) {
```

Comment:
Could there be a ``lastControlPoint()`` method on ``PathNode`` to describe this behavior / not repeat it for TtSs? I would imagine it just returning undefined by default and being overridden in ``QuadraticCurveTo`` and ``CubicCurveTo``